src/Symfony/Component/EventDispatcher/InvokableEventSubscriberTrait.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\EventDispatcher;
+
+use Symfony\Contracts\EventDispatcher\Event as ContractsEvent;
+
+/**
+ * Helper trait for single-event subscription.
+ *
+ * @author Yonel Ceruto <yonelceruto@gmail.com>
+ */
+trait InvokableEventSubscriberTrait
```

Comment:
`EventListenerTrait`? The emphasis should be on "Listener", not "Subscriber" IMHO.

Reply:
For the better, see discussion.

Reply:
yes and no.. we teach about listeners and subscribers since forever AFAIK: https://symfony.com/doc/current/event_dispatcher.html

i thought `extends EventListener` and `implements EventSubscriberInterface` reflected that pretty nice :) while technically we utilize around subscribers and couple with framework events either way

Reply:
see https://github.com/symfony/symfony/pull/33485#issuecomment-528795963

Reply:
i still think one chooses between a "single event listener" vs. many / an event subscriber beforehand, like a pattern decision. So " as soon as you want to listen to many events" does not apply IMHO.

Coupling by inheritance is still SRP compliant in this case.

Reply:
i figure for the component we're probably better off to not implement an event subscriber in disguise and calling it a listener, the initial RFC fits the framework's auto-configuration mostly.

Reply:
> we're probably better off to not implement an event subscriber

you mean: we add a base class that doesn't implement EventSubscriberInterface but is autoconfigured to add some new tag, and the new tag triggers a compiler pass that inspects the corresponding classes using reflection?

Honestly, that's a lot of complexity + concepts to teach, understand, mix with others. This trait achieves the same without any new concept. Adding all this just for saving `implements EventSubscriberInterface` looks like overkill to me. A few core simple powerful concepts are what ppl need to learn and use to me. Adding `implements EventSubscriberInterface` might act as a nice way to empower ppl into discovering its full semantics btw.

Reply:
no i mean `abstract EventListener implements EventSubscriberInterface` seems fine in the framework bundle, e.g. when writing end-user application listeners: `class Something extends EventListener + invoke` and done.

That would rely on existing instanceof/auto-configuration, and hide the fact we use an subscriber disguised as listener. However for the component i agree the trait approach is better, as doing `$dispatcher->addSubscriber($listener)` seems weird.

Reply:
we don't disguise a subscriber as listener. If your class knows to which event it want to listen, this is a Subscriber. A listener is just a callable.

I think that using `EventListener` as base class for a single-listener subscriber is a bad idea btw, as it will make it even more complex to explain the different between a listener and a subscriber (as this base class will make you implement a subscriber).